client/jetpack-connect/test/authorize.js

```diff
@@ -304,46 +354,107 @@ describe( 'JetpackAuthorize', () => {
 	} );
 
 	describe( 'getRedirectionTarget', () => {
-		test( 'should redirect to pressable if partnerSlug is "pressable"', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.setProps( {
-				partnerSlug: 'pressable',
-			} );
-			const target = component.instance().getRedirectionTarget();
+		let originalWindowLocation;
+
+		beforeEach( () => {
+			originalWindowLocation = global.window.location;
+			delete global.window.location;
+			global.window.location = {
+				href: 'http://wwww.example.com',
+				origin: 'http://www.example.com',
+			};
+		} );
+
+		afterEach( () => {
+			global.window.location = originalWindowLocation;
+		} );
+
+		test( 'should redirect to pressable if partnerSlug is "pressable"', async () => {
+			renderWithRedux(
+				<JetpackAuthorize
+					{ ...DEFAULT_PROPS }
+					authQuery={ {
+						...DEFAULT_PROPS.authQuery,
+						alreadyAuthorized: true,
+					} }
+					partnerSlug="pressable"
+					isAlreadyOnSitesList={ true }
+					isFetchingSites={ true }
+				/>
+			);
+
+			await userEvent.click( screen.getByText( 'Return to your site' ) );
+
+			const target = global.window.location.href;
```

Comment:
Since we're checking the actual value of the `href` property (which is what is set directly by the given logic branch in the `navigate` function, see [#](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/navigate/index.ts#L14)), I don't understand how using `toHaveBeenCalledWith` here would work in this case 🤔. Could you provide an example?

Reply:
Oh, I think I understand your confusion. I achieved this by using mocks with [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) and [setters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set) in a recent Gutenberg RTL migration. See #44470 and here's the actual mock [being declared](https://github.com/WordPress/gutenberg/blob/a3036828a41b0be8c79a6e7c459160d75cc018dd/packages/editor/src/components/post-preview-button/test/index.js#L30). At that point, it's easy enough to use `toHaveBeenCalledWith` on the inner mock that we invoke when the `location` or `href` setter is called. Let me know it that makes sense!

Reply:
That's neat! Thanks for the example! However, I think that for this specific test, the current approach ends up being more suitable/simpler because we still need to `delete` the `location` property to re-assign it, so `mockRestore` won't work on it, negating some (if not most) of the benefits of using the mock. I'm happy to further discuss and revisit it in a follow-up, though!